### PR TITLE
[macOS] Image Aspect Ratio Implementation

### PR DESCRIPTION
**Pull request by MarkoBL**
### Description of Change ###

This PR implements image aspect ratios for macOS. NSImageView doesn't support AspectFill, like iOS. Therefore, I used a NSView with a CALayer to mimic a NSImageView.

### Bugs Fixed ###

- Provide links to bugs here

### Behavioral Changes ###

Before this PR, the macOS platform didn't have image aspect ratios implemented.
### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Consolidate commits as makes sense

### CODE REVIEW

**Comment by rmarinho** [comment]
Why don't do it on the renderer itself ?

**Comment by MarkoBL** [comment]
Like this?

**Comment by rmarinho** [comment]
Almost, but why did you change to  `VisualElementRenderer<Image>` ? can you keep `ViewRenderer<Image,ImageView> `

File: Xamarin.Forms.Platform.MacOS/Renderers/ImageRenderer.cs [hunk 1]
```diff
@@ -1,42 +1,50 @@
 ﻿using System;
 using System.ComponentModel;
 using AppKit;
+using CoreAnimation;
+using CoreGraphics;
 
 namespace Xamarin.Forms.Platform.MacOS
 {
-	public class ImageRenderer : ViewRenderer<Image, NSImageView>
+	public class ImageRenderer : VisualElementRenderer<Image>
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.MacOS/Renderers/ImageRenderer.cs]: Can we keep the other version ?

**Comment by BradChase2011** [comment]
@rmarinho Where are we on this one?  Are there still changes that need to be done?

**Comment by rmarinho** [comment]
You should keep ViewRenderer @BradChase2011

**Comment by rmarinho** [comment]
Please reopen when you can work on this @MarkoBL

**Comment by BradChase2011** [comment]
@rmarinho @MarkoBL Mark I can take this over when I have time if you dont.  Its up to you.  If this goes stale then I can as well.  Currently we have a work around because we use all SVGs and have our own control that inherits from Image but I would like to have a final solution.

**Comment by MarkoBL** [comment]
I would like to finish this pull request. How can I reopen it?

**Comment by opcodewriter** [comment]
I thin you can't reopen it because @rmarinho closed it

**Comment by rmarinho** [comment]
@MarkoBL just rebase and push changes to your branch ..

**Comment by MarkoBL** [comment]
Sorry, I just created a new PR: https://github.com/xamarin/Xamarin.Forms/pull/1041

**Pull request closed without merging**